Quotient/e2ee/qolminboundsession.h

```diff
@@ -49,8 +49,9 @@ class QUOTIENT_API QOlmInboundGroupSession
     OlmErrorCode lastErrorCode() const;
     const char* lastError() const;
 
-private:
+    //TODO: why?
```

Comment:
Because so far all `QOlmInboundGroupSession` objects were supposed to be created by static factory methods. If that's no more the case, feel free to change it - but you don't seem to need opening it up so far?

Reply:
I'm still only using a factory method; the code fails to compile when the constructor is private, though

Reply:
Well, the current `Room::addMegolmSessionFromBackup()` code obviously tries to default-construct `QOlmInboundGroupSession` each time you try to access the session via `operator[]`. As I'm already hacking on this PR, I think I can rewrite it in a more idiomatic C++, with iterators and so.